src/Carbon/Traits/Localization.php

```diff
@@ -235,7 +235,7 @@ public static function translateWith(TranslatorInterface $translator, string $ke
             : $translator->transChoice($key, $number, $parameters);
         // @codeCoverageIgnoreEnd
 
-        return (string) $choice;
+        return $choice;
```

Comment:
Not guaranteed with symfony/translation 4

Reply:
reverted this one